contrib/cirrus/setup_environment.sh

```diff
@@ -19,6 +19,16 @@ die_unknown() {
     die "Unknown/unsupported \$$var_name '$var_value'"
 }
 
+# This branch is tied to RHEL 8.4.0.2 with an explicit EOL date.
+if [[ $(date +%s) -ge $(date -d "10/01/2023" +%s) ]]; then
+    die "This branch is probably no longer supported in RHEL,
+please confirm with PM.  If so:
+It should be removed from Cirrus-Cron,
+the .cirrus.yml file removed, and
+the VM images (manually) unmarked
+'permanent=true'"
+fi
+
```

Comment:
@edsantiago Quick opinion - is this a good idea?  Should I cherry-pick this to other branches (adjusting the date)?

Reply:
Clever! Yes, I think it's worth a try, although I wouldn't cherrypick it until it has a few weeks in production.

One suggestion inline, and one more: please include the strings `8.4.0.2` and `20231001` (can refactor the latter) in the output string. That makes it trivial for a human to read the error, determine that it's date-based, get the RHEL version string, and follow up from there without needing to read the source.